src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java

```diff
@@ -47,9 +47,11 @@
 public class RedisCacheConfiguration {
 
 	protected static final boolean DEFAULT_CACHE_NULL_VALUES = true;
+	protected static final boolean DEFAULT_ENABLE_TTI_EXPIRATION = false;
 	protected static final boolean DEFAULT_USE_PREFIX = true;
 	protected static final boolean DO_NOT_CACHE_NULL_VALUES = false;
 	protected static final boolean DO_NOT_USE_PREFIX = false;
+	protected static final boolean ENABLE_IDLE_TIME_EXPIRATION = true;
```

Comment:
@quaff has a good point regarding server versions. We cannot assume the application runs on the latest Redis version. If we'd enabled this for everyone, then all users with older Redis versions suddenly see failures on their target environments unless they reconfigure the cache.

Reply:
I did similar work before, like this:

```java

	private volatile boolean getexSupported = true;

	@Override
	public Object getWithTti(String key, int timeToIdle, TimeUnit timeUnit) {
		BoundValueOperations operations = redisTemplate.boundValueOps(key);
		try {
			if (getexSupported)
				return timeToIdle > 0 ? operations.getAndExpire(timeToIdle, timeUnit) : operations.getAndPersist();
			// fallback to get and expire
			Object result = operations.get();
			if (result != null && timeToIdle > 0)
				operations.expire(timeToIdle, timeUnit);
			return result;
		} catch (RuntimeException e) {
			log.error(e.getMessage(), e);
			if (e.getCause() instanceof RedisCommandExecutionException) {
				getexSupported = false; // server.version < 6.2
				return getWithTti(key,  timeToIdle, timeUnit);
			}
			throw e;
		}
	}
```

Reply:
Please note, the `ENABLE_IDLE_TIME_EXPIRATION` constant is **not** a flag defaulting _time-to-idle_ (TTI) expiration to `true`.

I simply prefer not to use constant values like  `true` for boolean flag method arguments, especially when or where constructors and/or methods accept multiple boolean parameters.  For [instance](https://github.com/spring-projects/spring-data-redis/blob/issue/2351/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L226), or [here](https://github.com/spring-projects/spring-data-redis/blob/issue/2351/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L192) and [here](https://github.com/spring-projects/spring-data-redis/blob/issue/2351/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L206).

However, I recognize the constant name (`ENABLE_IDLE_TIME_EXPIRATION `) is a really poor choice of words and have changed this to `USE_TIME_TO_IDLE_EXPIRATION`.

Keep in mind, this constant (now `USE_TIME_TO_IDLE_EXPIRATION`) is only ever used when TTI is enabled by calling the `enableTtiExpiration()` method on the `RedisCacheConfiguration` instance, [here](https://github.com/spring-projects/spring-data-redis/blob/issue/2351/src/main/java/org/springframework/data/redis/cache/RedisCacheConfiguration.java#L209-L229).

Reply:
If @christophstrobl and I decide it makes sense to handle Redis server version for a (Redis) `Cache.get(key)`, using either `GETEX` or a `GET` with `EXPIRE` (although, I am not certain SD Redis takes Redis server version into consideration anywhere else, though), then I would envision something a bit more robust, such as by (proactively) inspecting the Redis server version on construction of the `DefaultRedisCacheWriter` using the `INFO` command from `RedisServerCommands` given a `RedisConnection` (retrieved from the factory).